botorch/utils/probability/mvnxpb.py

```diff
@@ -176,18 +176,20 @@ def solve(self, num_steps: Optional[int] = None, eps: float = 1e-10) -> Tensor:
             if pivot is not None and torch.any(pivot > i):
                 self.pivot_(pivot=pivot)
 
-            # Initialize `i`-th plug-in value as univariate conditional expectation
+            # Compute whitened bounds conditional on preceding plug-ins
             Lii = L[..., i, i].clone()
             if should_update_chol:
-                Lii = Lii.clip(min=0).sqrt()
+                Lii = Lii.clip(min=0).sqrt()  # conditional stddev
             inv_Lii = Lii.reciprocal()
-            if i == 0:
-                lb, ub = bounds[..., i, :].clone().unbind(dim=-1)
-            else:
-                db = (L[..., i, :i].clone() * y[..., :i].clone()).sum(-1, keepdim=True)
-                lb, ub = (bounds[..., i, :].clone() - db).unbind(dim=-1)
-
-            Phi_i = Phi(inv_Lii * ub) - Phi(inv_Lii * lb)
+            bounds_i = bounds[..., i, :].clone()
+            if i != 0:
+                bounds_i = bounds_i - torch.einsum(
+                    "...i,...i", L[..., i, :i].clone(), y[..., :i].clone()
+                ).unsqueeze(-1)
```

Comment:
Why change to `einsum` here? The parsing of the formula actually has some overhead - I just tested this, for small dimensions the previous implementation is about 5X faster.

Reply:
Ah, good catch. I switched to `einsum` to avoid allocating additional memory, thinking that it would run at roughly the same speed. Since it's significantly slower, we should probably revert.